pkg/vault/test/foundry/utils/BasePoolTest.sol

```diff
@@ -0,0 +1,295 @@
+            vault.manuallySetSwapFee(pool, 0);
+        }
+
+        IERC20 tokenIn = poolTokens[tokenIndexIn];
+        IERC20 tokenOut = poolTokens[tokenIndexOut];
+
+        vm.prank(bob);
+        uint256 amountCalculated = router.swapSingleTokenExactIn(
+            pool,
+            tokenIn,
+            tokenOut,
+            tokenAmountIn,
+            _less(tokenAmountOut, 1e3),
+            MAX_UINT256,
+            false,
+            bytes("")
+        );
+
+        // Tokens are transferred from Bob
+        assertEq(tokenOut.balanceOf(bob), defaultBalance + amountCalculated, "LP: Wrong tokenOut balance");
+        assertEq(tokenIn.balanceOf(bob), defaultBalance - tokenAmountIn, "LP: Wrong tokenIn balance");
+
+        // Tokens are stored in the Vault
+        assertEq(
+            tokenOut.balanceOf(address(vault)),
+            tokenAmounts[tokenIndexOut] - amountCalculated,
+            "Vault: Wrong tokenOut balance"
+        );
+        assertEq(
+            tokenIn.balanceOf(address(vault)),
+            tokenAmounts[tokenIndexIn] + tokenAmountIn,
+            "Vault: Wrong tokenIn balance"
+        );
+
+        (, , uint256[] memory balances, ) = vault.getPoolTokenInfo(pool);
+
+        assertEq(balances[tokenIndexIn], tokenAmounts[tokenIndexIn] + tokenAmountIn, "Pool: Wrong tokenIn balance");
+        assertEq(
+            balances[tokenIndexOut],
+            tokenAmounts[tokenIndexOut] - amountCalculated,
+            "Pool: Wrong tokenOut balance"
+        );
+    }
+
+    function testMinimumSwapFee() public view {
+        assertEq(IBasePool(pool).getMinimumSwapFeePercentage(), MIN_SWAP_FEE, "Minimum swap fee mismatch");
+    }
+
+    function testMaximumSwapFee() public view {
+        assertEq(IBasePool(pool).getMaximumSwapFeePercentage(), MAX_SWAP_FEE, "Maximum swap fee mismatch");
+    }
+
+    function testSetSwapFeeTooLow() public {
+        authorizer.grantRole(vault.getActionId(IVaultAdmin.setStaticSwapFeePercentage.selector), alice);
+        vm.prank(alice);
+
+        vm.expectRevert(IVaultErrors.SwapFeePercentageTooLow.selector);
+        vault.setStaticSwapFeePercentage(pool, MIN_SWAP_FEE - 1);
+    }
+
+    function testSetSwapFeeTooHigh() public {
+        authorizer.grantRole(vault.getActionId(IVaultAdmin.setStaticSwapFeePercentage.selector), alice);
+        vm.prank(alice);
+
+        vm.expectRevert(abi.encodeWithSelector(IVaultErrors.SwapFeePercentageTooHigh.selector));
+        vault.setStaticSwapFeePercentage(pool, MAX_SWAP_FEE + 1);
+    }
+
+    function testAddLiquidityUnbalanced() public {
+        authorizer.grantRole(vault.getActionId(IVaultAdmin.setStaticSwapFeePercentage.selector), alice);
+        vm.prank(alice);
+        vault.setStaticSwapFeePercentage(pool, 10e16);
+
+        uint256[] memory amountsIn = new uint256[](poolTokens.length);
+        amountsIn[0] *= 100;
+        vm.prank(bob);
+
+        router.addLiquidityUnbalanced(pool, amountsIn, 0, false, bytes(""));
+    }
+
```

Comment:
Wasn't in the original, but could add creation below the minimum fee here, too. I suppose it could be in the BasePoolTest, but would be a bit messy, as you'd have to parameterize `createPool`, so probably not worth it.

```
    function testFailSwapFeeTooLow() public {
        TokenConfig[] memory tokenConfigs = new TokenConfig[](2);
        for (uint256 i = 0; i < poolTokens.length; i++) {
            tokenConfigs[i].token = poolTokens[i];
        }

        PoolRoleAccounts memory roleAccounts;

        address lowFeeStablePool = address(
            StablePool(
                StablePoolFactory(address(factory)).create(
                    "ERC20 Pool",
                    "ERC20POOL",
                    tokenConfigs,
                    DEFAULT_AMP_FACTOR,
                    roleAccounts,
                    MIN_SWAP_FEE - 1,
                    poolHooksContract,
                    false, // Do not enable donations
                    false, // Do not disable unbalanced add/remove liquidity
                    ZERO_BYTES32
                )
            )
        );

        factoryMock.registerTestPool(lowFeeStablePool, tokenConfigs);
    }
```